src/Composer/Util/Zip.php

```diff
@@ -76,27 +78,32 @@ private static function locateFile(\ZipArchive $zip, $filename)
             return $index;
         }
 
+        $foundFileCount = 0;
         $topLevelPaths = array();
         for ($i = 0; $i < $zip->numFiles; $i++) {
             $name = $zip->getNameIndex($i);
             $dirname = dirname($name);
+            $stat = $zip->statIndex($i);
+            if (basename($stat['name']) === $filename) {
+                $foundFileCount++;
+            }
 
             // handle archives with proper TOC
             if ($dirname === '.') {
                 $topLevelPaths[$name] = true;
                 if (\count($topLevelPaths) > 1) {
-                    // archive can only contain one top level directory
-                    return false;
```

Comment:
I think it should always throw here, not keep track of `foundFileCount`. The error here is that the archive has more than one top level directories, and no composer.json was found on the top level, so it's an invalid archive. The amount of composer.json does not matter and we should not keep track of it.

Reply:
ok for updating this and thorwing an exception, but shouldn't I keep the count on line 115?

Reply:
I don't think it's an error to have multiple composer.json files.. as the code here checks that there is only one top level directory, you can't possibly end up with two composer.json being `a/composer.json` and `b/composer.json` which would be ambiguous. It would already error when it sees `a/` and `b/` (we can even add the `$topLevelPaths` contents to the exception message to make it very clear what the issue is).